### PR TITLE
Added Fix as proposed in #6 and added Fixed Mode and Textcolor

### PR DESCRIPTION
**Pull request by escamoteur**
#6 Added the proposed fix

### CODE REVIEW

**Comment by escamoteur** [comment]
Added a property FixedMode to BottombarPage so that you now can also use FixedMode.

When selecting FixedMode also the TabText and Icon color can be set by setting BarTextColor

*You can only select a the Text and Icon Color of the active Tab when using Fixed Mode*

**Comment by andreinitescu** [comment]
Can you please remove the changes which are not really part of the PR? 
Please just keep the actual code, not changes due to different IDE.
Just look to the changes of `BottomBar.Droid/BottomBar.Droid.csproj` and `BottomBarXF.sln` to see what I mean.
Thanks!

File: BottomBar.XamarinForms/BottomBarPage.cs [hunk 1]
```diff
@@ -21,7 +21,9 @@ namespace BottomBar.XamarinForms
 {
 	public class BottomBarPage : TabbedPage
 	{
-		public void RaiseCurrentPageChanged()
+        public bool FixedMode { get; set; }
```
Comment (andreinitescu) [inline BottomBar.XamarinForms/BottomBarPage.cs]: code is not formatted....

File: BottomBar.Droid/Renderers/BottomBarPageRenderer.cs [hunk 2]
```diff
@@ -132,7 +132,11 @@ protected override void OnElementChanged (ElementChangedEventArgs<BottomBarPage>
 					// create bottomBar control
 					_bottomBar = BottomNavigationBar.BottomBar.Attach (_frameLayout, null);
 					_bottomBar.NoTabletGoodness ();
-					_bottomBar.LayoutParameters = new LayoutParams (LayoutParams.MatchParent, LayoutParams.MatchParent);
+				    if (bottomBarPage.FixedMode)
```
Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Could you please format the code?

File: BottomBar.Droid/Renderers/BottomBarPageRenderer.cs [hunk 3]
```diff
@@ -227,7 +231,8 @@ void UpdateBarTextColor ()
 				return;
 			}
 
-			// haven't found yet how to set text color for tab items on_bottomBar, doesn't seem to have a direct way
+            _bottomBar.SetActiveTabColor(Element.BarTextColor.ToAndroid());
```
Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: same here...

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Strange it's formated at my side.

Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Doesn't appear so: https://github.com/escamoteur/BottomNavigationBarXF/blob/b07806b2d7bf57fd348e12720ea3d39f5a97fcb9/BottomBar.Droid/Renderers/BottomBarPageRenderer.cs#L234

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: That's strange. must be another problem.
I create a new branch

Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: not sure, 
maybe create a new branch and just push PR from it with only code which is changed?
not sure how will we handle the situation when a new file is added to the .csproj....
another idea would be to just go ahead with the PR and stick to Visual Studio from now on, but it would require some reformatting of whole code...

Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: at the time I wrote this, I think so

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: and the others on the project?And the other ones on the project? Normally it should not be a problem between XS and VS I work on another OSS project where it's not a problem

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Couldn't you just merge this PR and exclude the project files. I don't know what I could do with the formating as it's formated here

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: thanks!

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: And please may I ask for a bit more friendly reactions? Because as in my case I was not aware of any formating problems.

Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: I'm sorry, friendly reactions? what do you mean

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: This is my first PR to this project and I felt you comments quite harsh as if I'm an idiot who cannot format his code before pushing.

Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: I'm sorry if you feel that way. Please read again, I didn't have any harsh comments, I just asked if you could remove the unnecessary changes, which add a lot of noise to the PR.

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Don't want to push this further. Perhaps the next time first welcome a new contributor. It's about motivation to contribute on a project.
I'm fine now, we found the reason.

**Comment by escamoteur** [comment]
This is strange, when using FixedMode I can set the TextColor, but no longer the Bar Color. any idea?

**Comment by escamoteur** [comment]
Ok, found out how it works. You have to set DarkTheme and then provide a colors.xml file. I added a property for the theming to BottomBarPage

File: BottomBar.Droid/Renderers/BottomBarPageRenderer.cs (hunk 3 above)
Comment (andreinitescu) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: I didn't welcome you? I'm sorry for that. Welcome again!

Comment (escamoteur) [inline BottomBar.Droid/Renderers/BottomBarPageRenderer.cs]: Ahh written communication is so difficult if it comes to emotions :-(

**Comment by escamoteur** [comment]
Found today that the CurrentPageChangedEvent was called twice because it was called due to the asignment of the new Page to CurrentPage which triggers an OnPropertyChanged event and on the following call to RaiseCurrentPageChanged.

I changed it to

```
		public void OnTabSelected (int position)
		{
            //Do we need this call? It's also done in OnElementPropertyChanged
            SwitchContent(Element.Children [position]);
			var bottomBarPage = Element as BottomBarPage;
			bottomBarPage.CurrentPage = Element.Children[position];
			//bottomBarPage.RaiseCurrentPageChanged();
		}
```

As I wrote in the comment I'm not sure if we need the first call to SwitchContent as it get called again when the CurrentPage property get's changed:

```
		protected override void OnElementPropertyChanged (object sender, PropertyChangedEventArgs e)
		{
			base.OnElementPropertyChanged (sender, e);

			if (e.PropertyName == nameof (TabbedPage.CurrentPage)) {
				SwitchContent (Element.CurrentPage);
			} else if (e.PropertyName == NavigationPage.BarBackgroundColorProperty.PropertyName) {
				UpdateBarBackgroundColor ();
			} else if (e.PropertyName == NavigationPage.BarTextColorProperty.PropertyName) {
				UpdateBarTextColor ();
			}
		}
```
If you comment it out it still works.

**Comment by escamoteur** [comment]
@sschmidt did you have a loom at my PR?

**Comment by nbsoftware** [comment]
Hello @escamoteur !
Thanks for your contributions ;)
Actually you are right, I also just found out about the `CurrentPageChangedEvent` being called twice.
So your fix is correct, I did the same and I actually also removed the line that calls SwitchContent in `OnTabSelected` because it is indeed done in `OnElementPropertyChanged`.
So the function looks like that:
```
		public void OnTabSelected (int position)
		{
			var bottomBarPage = Element as BottomBarPage;
			// setting the CurrentPage on XF control will raise the corresponding OnElementPropertyChanged call
			// so no need to call SwitchContent from here
			bottomBarPage.CurrentPage = Element.Children[position];
		}
```

While I am at it I made another fix in `SwitchContent` for the following scenario: if the CurrentPage property was set programmatically in the PCL code, the correct tab view was indeed updated and shown, but not the corresponding button. Fixed by calling `SelectTabAtPosition` on native control.
```

	protected virtual void SwitchContent (Page view)
		{
			// when CurrentPage has been changed programmatically from Xamarin Form control,
			// we need to set the current active tab also on the native control
			_bottomBar.SelectTabAtPosition(Element.Children.IndexOf(Element.CurrentPage), false);

			Context.HideKeyboard (this);

			_frameLayout.RemoveAllViews ();

			if (view == null) {
				return;
			}

			if (Platform.GetRenderer (view) == null) {
				Platform.SetRenderer (view, Platform.CreateRenderer (view));
			}

			_frameLayout.AddView (Platform.GetRenderer (view).ViewGroup);
		}

```

If you want I can commit those changes to my master branch and send you the PR...
Or, as they are really simple, you could apply those 2 changes to yours directly.

Cheers

**Comment by escamoteur** [comment]
Great, what about the other extensions? what do you think about them?

**Comment by sschmidt** [comment]
Hi @escamoteur, thanks for the great contribution. Highly appreciated!

Do we really need the changes to the sln and csproj files (i.e. changes to visual studio version/target framework version)? In case those changes don't add to the solution, I'd like to leave those files as they are. 

As I understand @nbsoftware  and @andreinitescu already approved/tested this PR (thanks guys!). I'll test it tomorrow and we can go ahead and merge this in.

**Comment by escamoteur** [comment]
Sorry, this seems to be an effect of VS 2015 with Xamarin 4.2. You don't need to merge them in.

**Comment by escamoteur** [comment]
Great, thanks. Do you built a new nuget?

**Comment by sschmidt** [comment]
Sure, can you briefly test the master version before I update nuget please?

**Comment by nbsoftware** [comment]
Hello,

Before releasing to nuget, could you please integrate the two small fixes I mentioned in https://github.com/thrive-now/BottomNavigationBarXF/pull/16#issuecomment-254543261 ?

Would you prefer I issue a PR from current master?

Thanks

**Comment by sschmidt** [comment]
@nbsoftware sure. A PR would be more than welcome!

**Comment by escamoteur** [comment]
Did anybody of you try to change the fond/fontsize of the Tabb Texts in fixed mode? One of my texts get's truncated while perfectly visible on iOS.

**Comment by escamoteur** [comment]
Just checked the current version works fine.
